src/test/java/com/fasterxml/jackson/databind/util/TokenBufferTest.java

```diff
@@ -194,6 +194,32 @@ public void testNumberOverflowLong() throws IOException
         }
     }
 
+    public void testBigIntAsString() throws IOException
+    {
+        try (TokenBuffer buf = new TokenBuffer(null, false)) {
+            BigInteger big = BigInteger.valueOf(Long.MAX_VALUE).add(BigInteger.valueOf(1));
+            buf.writeNumber(big.toString());
+            try (JsonParser p = buf.asParser()) {
+                // NOTE: oddity of buffering, no inspection of "real" type if given String...
```

Comment:
I think we can actually solve that in near future, by buffering number-as-Strings as deferred value that retains type information (at very least integer/fp distinction)